manheim_c7n_tools/dryrun_diff.py

```diff
@@ -100,6 +109,67 @@ def _find_changed_policies(self, git_dir=None, diff_against='master'):
             pnames.append(m.group(1))
         return pnames
 
+    def _make_diff_report(self, dryrun):
+        """
+        Return a HTML report breaking down the differences between the dryrun
+        (this branch) and the last run of each policy on master.
+
+        :param dryrun: dryrun policy resource information
+        :type dryrun: dict
+        :return: html report
+        :rtype: str
+        """
+        all_policies = list(
+            set(dryrun.keys()) | set(self._live_results.keys())
+        )
+        entries = {}
+        if not all_policies:
+            logger.info('no policies found - skipping diff report')
+            return ''
+        for policy in all_policies:
+            for region in self.config.regions:
+                dry_ids = {
+                    x['Arn']
```

Comment:
There is a big assumption here regarding the availablity of the `Arn` key... perhaps I'll think of something more clever to extract an ID based on a resource type

Reply:
Apologies for the delay reviewing this... I'll get to it next week.

I don't remember offhand, but I'm pretty sure that the c7n data should have a resource ID key in it? If not, there's definitely code in upstream c7n to get that...

Reply:
Good shout, I've opted for the [following imports](https://github.com/manheim/manheim-c7n-tools/pull/34/commits/c9b1b8b77a1c7f16b632d22160fa80d3d12f910c#diff-12aec96e72fb0486abb7d120ef81c3d7R35-R36) from upstream:

```
from c7n.resources import load_resources
from c7n.provider import cloud
```

I'm able to manipuate both packages to generate a ResourceManager for a given `resource_type` which I grab from the prefixes for policies (not sure if that's too big of an assumption). I can then query the ResourceManager to give me the `id` for the associated type; i.e for an `ec2` instance, it'd return `InstanceId`